raiden/tasks.py

```diff
@@ -134,6 +135,78 @@ def stop_async(self):
         self.stop_event.set(True)
 
 
+class HealthcheckTask(Task):
+    """ Task for checking if all of our open channels are healthy """
+
+    def __init__(
+            self,
+            raiden,
+            send_ping_time,
+            max_unresponsive_time,
+            sleep_time=DEFAULT_HEALTHCHECK_POLL_TIMEOUT):
+        """ Initialize a HealthcheckTask that will monitor open channels for
+             responsiveness.
+
+             :param raiden RaidenService: The Raiden service which will give us
+                                          access to the protocol object and to
+                                          the asset manager
+             :param int sleep_time: Time in seconds between each healthcheck task
+             :param int send_ping_time: Time in seconds after not having received
+                                        a message from an address at which to send
+                                        a Ping.
+             :param int max_unresponsive_time: Time in seconds after not having received
+                                               a message from an address at which it
+                                               should be deleted.
+         """
+        super(HealthcheckTask, self).__init__()
+
+        self.protocol = raiden.protocol
+        self.raiden = raiden
+
+        self.stop_event = AsyncResult()
+        self.sleep_time = sleep_time
+        self.send_ping_time = send_ping_time
+        self.max_unresponsive_time = max_unresponsive_time
+
+    def _run(self):  # pylint: disable=method-hidden
+        stop = None
+        while stop is None:
+            keys_to_remove = []
+            for key, queue in self.protocol.address_queue.iteritems():
```

Comment:
@hackaugusto That is a considerable re-work of the healthcheck task as it is now. I am not opposed to it, but I would rather this suggestion be an issue to be handled on top of the current healthcheck task in another PR in the future.

We should take this discussion to [an issue](https://github.com/raiden-network/raiden/issues/247) and that issue should become a PR to improve the first healthcheck task implementation.